src/bin/pgcopydb/copydb.c

```diff
@@ -633,12 +633,12 @@ copydb_init_specs(CopyDataSpec *specs,
 		.cfPaths = specs->cfPaths,
 		.pgPaths = specs->pgPaths,
 
-		.source_pguri = { 0 },
-		.target_pguri = { 0 },
+		.source_pguri = (char *) calloc(strlen(options->source_pguri) + 1, sizeof(char)),
+		.target_pguri = (char *) calloc(strlen(options->target_pguri) + 1, sizeof(char)),
 
 		.sourceSnapshot = {
 			.pgsql = { 0 },
-			.pguri = { 0 },
+			.pguri = (char *) calloc(strlen(options->source_pguri) + 1, sizeof(char)),
```

Comment:
Is there a way that we can share the existing memory area and just copy the pointer to it here?

Reply:
Otherwise again, just use strdup.